pisek/task_jobs/generator/manager.py

```diff
@@ -0,0 +1,220 @@
+from pisek.utils.paths import TaskPath
+from pisek.config.config_types import GenType, DataFormat
+from pisek.jobs.jobs import Job, JobManager
+from pisek.task_jobs.task_manager import TaskJobManager
+from pisek.task_jobs.compile import Compile
+from pisek.task_jobs.data.data import InputSmall
+from pisek.task_jobs.tools import IsClean
+from pisek.task_jobs.checker import CheckerJob
+
+from .input_info import InputInfo
+from .base_classes import (
+    GeneratorListInputs,
+    GenerateInput,
+    GeneratorTestDeterminism,
+    GeneratorRespectsSeed,
+)
+from .cms_old import CmsOldListInputs, CmsOldGenerate
+from .opendata_v1 import (
+    OpendataV1ListInputs,
+    OpendataV1Generate,
+    OpendataV1TestDeterminism,
+)
+
+SEED_RANGE = range(0, 16**8)
+
+
+class PrepareGenerator(TaskJobManager):
+    """Manager that generates inputs and test generator."""
+
+    def __init__(self):
+        self._inputs = []
+        super().__init__("Prepare generator")
+
+    def _get_jobs(self) -> list[Job]:
+        generator = self._env.config.in_gen
+
+        jobs: list[Job] = [
+            compile_gen := Compile(self._env, generator),
+            list_inputs := list_inputs_job(self._env, generator),
+        ]
+        list_inputs.add_prerequisite(compile_gen)
+        self._list_inputs = list_inputs
+
+        return jobs
+
+    def _compute_result(self) -> dict[str, Any]:
+        return {"inputs": self._list_inputs.result}
+
+
+def list_inputs_job(env: Env, generator: TaskPath) -> GeneratorListInputs:
+    return {
+        GenType.opendata_v1: OpendataV1ListInputs,
+        GenType.cms_old: CmsOldListInputs,
+    }[env.config.gen_type](env=env, generator=generator)
+
+
+def generate_input(
+    env: Env, generator: TaskPath, input_info: InputInfo, seed: int
+) -> GenerateInput:
+    return {
+        GenType.opendata_v1: OpendataV1Generate,
+        GenType.cms_old: CmsOldGenerate,
+    }[
+        env.config.gen_type
+    ](env=env, generator=generator, input_info=input_info, seed=seed)
+
+
+def generator_test_determinism(
+    env: Env, generator: TaskPath, input_info: InputInfo, seed: int
+) -> Optional[GeneratorTestDeterminism]:
+    TEST_DETERMINISM = {GenType.opendata_v1: OpendataV1TestDeterminism}
+
+    if env.config.gen_type not in TEST_DETERMINISM:
+        return None
+    return TEST_DETERMINISM[env.config.gen_type](
+        env=env, generator=generator, input_info=input_info, seed=seed
+    )
+
+
+class InputsInfoMixin(JobManager):
+    def __init__(self, name: str, **kwargs) -> None:
+        self._generate_inputs: dict[int, GenerateInput] = {}
+        super().__init__(name=name, **kwargs)
+
+    def _input_info_jobs(self, input_info: InputInfo, subtask: int) -> list[Job]:
+        repeat = input_info.repeat * (
+            self._env.repeat_inputs if input_info.seeded else 1
+        )
+
+        jobs: list[Job] = []
+
+        random.seed(4)  # Reproducibility!
+        seeds = random.sample(SEED_RANGE, repeat)
```

Comment:
If there is any change of implementing parallelism in the future, it might be a good idea to construct a `Random` instance instead of changing global state.

Reply:
I would leave paralelism for the far far future and not bother with it now.

Reply:
I'd still prefer to avoid unneeded global state,  especially because it's a simple change (`random.Random(4).sample(...)`).